cpp/tests/reshape/byte_cast_tests.cpp

```diff
@@ -0,0 +1,343 @@
+  fixed_width_column_wrapper<float> const fp32_col(
+    {float(0.0), float(100.0), float(-100.0), limits::min(), limits::max(), nan, -nan, inf, -inf});
+  lists_column_wrapper<uint8_t> const fp32_expected({{0x00, 0x00, 0x00, 0x00},
+                                                     {0x00, 0x00, 0xc8, 0x42},
+                                                     {0x00, 0x00, 0xc8, 0xc2},
+                                                     {0x00, 0x00, 0x80, 0x00},
+                                                     {0xff, 0xff, 0x7f, 0x7f},
+                                                     {0x00, 0x00, 0xc0, 0x7f},
+                                                     {0x00, 0x00, 0xc0, 0x7f},
+                                                     {0x00, 0x00, 0x80, 0x7f},
+                                                     {0x00, 0x00, 0x80, 0xff}});
+  lists_column_wrapper<uint8_t> const fp32_expected_slice1({{0x00, 0x00, 0x00, 0x00},
+                                                            {0x42, 0xc8, 0x00, 0x00},
+                                                            {0xc2, 0xc8, 0x00, 0x00},
+                                                            {0x00, 0x80, 0x00, 0x00},
+                                                            {0x7f, 0x7f, 0xff, 0xff}});
+  lists_column_wrapper<uint8_t> const fp32_expected_slice2({{0x7f, 0xc0, 0x00, 0x00},
+                                                            {0x7f, 0xc0, 0x00, 0x00},
+                                                            {0x7f, 0x80, 0x00, 0x00},
+                                                            {0xff, 0x80, 0x00, 0x00}});
+
+  std::vector<cudf::size_type> splits({5});
+  std::vector<cudf::column_view> split_column = cudf::split(fp32_col, splits);
+
+  auto const output_fp32        = cudf::byte_cast(fp32_col, cudf::flip_endianness::NO);
+  auto const output_fp32_slice1 = cudf::byte_cast(split_column.at(0), cudf::flip_endianness::YES);
+  auto const output_fp32_slice2 = cudf::byte_cast(split_column.at(1), cudf::flip_endianness::YES);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output_fp32->view(), fp32_expected, true);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output_fp32_slice1->view(), fp32_expected_slice1, true);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output_fp32_slice2->view(), fp32_expected_slice2, true);
+}
+
+TEST_F(ByteCastTest, fp32ValuesWithNulls)
+{
+  using limits       = std::numeric_limits<float>;
+  auto even_validity = make_counting_transform_iterator(0, [](auto i) { return (i + 1) % 2; });
+
+  fixed_width_column_wrapper<float> const fp32_col(
+    {float(0.0), float(100.0), float(-100.0), limits::min(), limits::max()}, {1, 0, 1, 0, 1});
+  // auto fp32_data =
+  //   fixed_width_column_wrapper<uint8_t>{0x00, 0x00, 0x00, 0x00, 0xcc, 0xdd, 0xee, 0xff, 0xc2,
+  //   0xc8,
+  //                                       0x00, 0x00, 0xcc, 0xdd, 0xee, 0xff, 0x7f, 0x7f, 0xff,
+  //                                       0xff};
```

Comment:
Can this comment be removed? There are a few other comments similar to this above too.

Reply:
This is a placeholder for an issue regarding list comparison. Right now `CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT` does not compare nulls correctly and still expects the underlying nulled out values to be equal. The commented out values test that null case, but are commented out to allow tests to pass because testing that case is not possible right now.

Should they still be removed?

Reply:
That is fine but perhaps you could add a comment to why they are here then?

Reply:
Added in the first test set, should the same comment be added to every test set?

Reply:
Maybe just a 'see comment above' type comment.
Just worried someone will just go through and remove these lines without knowing why they are here.

Reply:
Done, block commenting used to avoid the commented code merging with the comment explaining it.